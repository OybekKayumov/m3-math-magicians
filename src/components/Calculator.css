.box-container {
  width: 100%;
  height: 60vh;
  background-color: #808080;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(6, 1fr);
}

.result-screen {
  grid-column: 1 / -1;
  color: #fff;
  padding: 2rem;
  display: flex;
  justify-content: flex-end;
  font-size: 2rem;
  background-size: cover;
}

.btn {
  all: unset;
  display: grid;
  place-content: center;
  background-color: #ccc;
  border: 1px solid #bbb;
  cursor: pointer;
  padding: 2rem 4rem;
  font-size: 1.4rem;
}

.btn:hover {
  background-color: #808080;
  color: #fff;
}

.btn:active {
  background-color: #000;
}

.hover-btn {
  background-color: #f3914e;
}

.hover-btn:hover {
  background-color: #ff4311;
}

.hover-btn:active {
  background-color: #ec4353;
}

.screen-num {
  grid-column: 1 / span 2;
}
