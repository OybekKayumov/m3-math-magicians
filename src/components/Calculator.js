import React, { Component } from 'react';
import './Calculator.css';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="box-container">
        <div className="result-screen">0</div>
        <button type="button" className="btn">AC</button>
        <button type="button" className="btn">+/-</button>
        <button type="button" className="btn">%</button>
        <button type="button" className="btn hover-btn">รท</button>
        <button type="button" className="btn">7</button>
        <button type="button" className="btn">8</button>
        <button type="button" className="btn">9</button>
        <button type="button" className="btn hover-btn">x</button>
        <button type="button" className="btn">4</button>
        <button type="button" className="btn">5</button>
        <button type="button" className="btn">6</button>
        <button type="button" className="btn hover-btn">-</button>
        <button type="button" className="btn">1</button>
        <button type="button" className="btn">2</button>
        <button type="button" className="btn">3</button>
        <button type="button" className="btn hover-btn">+</button>
        <button type="button" className="btn screen-num">0</button>
        <button type="button" className="btn">.</button>
        <button type="button" className="btn hover-btn">=</button>
      </div>
    );
  }
}

export default Calculator;
