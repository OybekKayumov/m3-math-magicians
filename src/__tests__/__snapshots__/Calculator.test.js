import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../../components/Calculator';

it('renders correctly when there are no items', () => {
  const calc = renderer.create(<Calculator />).toJSON();

  expect(calc).toMatchSnapshot();
});
