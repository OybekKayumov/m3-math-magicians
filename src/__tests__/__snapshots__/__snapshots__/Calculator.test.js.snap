// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="main-container"
>
  <p>
    Let's Do Some Math!
  </p>
  <div
    className="box-container"
  >
    <div
      className="screen"
    >
      0
    </div>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn hover-btn"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn hover-btn"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn hover-btn"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn hover-btn"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn screen-num"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn hover-btn"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
